src/sources/aws_s3/sqs.rs

```diff
@@ -475,12 +480,23 @@ impl IngestorProcess {
                         }
                     }
 
-                    ready(Some(log.into()))
+                    let event: Event = log.into();
+
+                    emit!(&SqsS3EventsReceived {
+                        byte_size: event.size_of()
+                    });
+
+                    ready(Some(event))
                 });
 
+                let (count, _) = stream.size_hint();
+
                 let send_error = match self.out.send_all(&mut stream).await {
                     Ok(_) => None,
-                    Err(_) => Some(crate::source_sender::ClosedError),
+                    Err(error) => {
+                        emit!(&StreamClosedError { error, count });
```

Comment:
I'm not sure we want to report discarded events here because an error here means we won't ack the SQS message and therefore would reprocess the events next time we fetch the message. In general, I think we should only report discarded events if the source does actually fully consume and ack the events, but still discards them.

Reply:
I opened a PR to clarify the component spec here: https://github.com/vectordotdev/vector/pull/11184

Reply:
I set the count to 0 in the latest commit so that the discarded counter is not incremented 😉